--- a/build.go
+++ b/build.go
@@ -32,8 +32,6 @@
 	"strings"
 	"text/template"
 	"time"
-
-	buildpkg "github.com/syncthing/syncthing/lib/build"
 	"github.com/syncthing/syncthing/lib/upgrade"
 	"sigs.k8s.io/yaml"
 )
@@ -414,13 +412,6 @@
 	}
 	args = append(args, "-timeout", timeout)
 
-	if runtime.GOARCH == "amd64" {
-		switch runtime.GOOS {
-		case buildpkg.Darwin, buildpkg.Linux, buildpkg.FreeBSD: // , "windows": # See https://github.com/golang/go/issues/27089
-			args = append(args, "-race")
-		}
-	}
-
 	if coverage {
 		args = append(args, "-covermode", "atomic", "-coverprofile", "coverage.txt", "-coverpkg", strings.Join(pkgs, ","))
 	}
